Lib/urllib/parse.py

```diff
@@ -166,7 +166,11 @@ def hostname(self):
     def port(self):
         port = self._hostinfo[1]
         if port is not None:
-            port = int(port, 10)
+            try:
+                port = int(port, 10)
+            except ValueError:
+                raise ValueError("Port could not be cast " \
```

Comment:
We could use an f-string here:

```py
f'Port could not be cast to integer value as {port!r}'
```

Reply:
Style nit: I'd either use a `msg` or `message` variable or a different indentation style instead of backslashes:

```py
message = f'Port could not be cast to integer value as {port!r}'
raise ValueError(message) from None
```

or

```py
raise ValueError(
    f'Port could not be cast to integer value as {port!r}'
) from None
```

Reply:
Will update.  Thank you!